modules/ml/src/main/java/org/apache/ignite/ml/math/distances/JaccardIndex.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.ml.math.distances;
+
+import java.util.HashSet;
+import java.util.Set;
+import org.apache.ignite.ml.math.exceptions.math.CardinalityException;
+import org.apache.ignite.ml.math.primitives.vector.Vector;
+
+/**
+ * Calculates {@code J = |A \cap B| / |A \cup B| } (Jaccard index) distance between two points.
+ */
+public class JaccardIndex implements DistanceMeasure {
+
+    @Override
+    public double compute(Vector a, Vector b) throws CardinalityException {
+        Set<Double> aSet = new HashSet<>();
```

Comment:
rename the variable to something meaningful

Reply:
It's hard :) I renamed `aSet` -> `uniqueValues`. If you have any better idea please, let me know.